app/reducers.js

```diff
@@ -2,49 +2,22 @@
  * Combine all reducers in this file and export the combined reducers.
  */
 
-import { fromJS } from 'immutable';
 import { combineReducers } from 'redux-immutable';
-import { LOCATION_CHANGE } from 'react-router-redux';
+import { connectRouter } from 'connected-react-router/immutable';
+import history from 'utils/history';
 
 import globalReducer from 'containers/App/reducer';
 import languageProviderReducer from 'containers/LanguageProvider/reducer';
 
-/*
- * routeReducer
- *
- * The reducer merges route location changes into our immutable state.
- * The change is necessitated by moving to react-router-redux@5
- *
- */
-
-// Initial routing state
-const routeInitialState = fromJS({
-  location: null,
-});
-
-/**
- * Merge route into the global application state
- */
-export function routeReducer(state = routeInitialState, action) {
-  switch (action.type) {
-    /* istanbul ignore next */
-    case LOCATION_CHANGE:
-      return state.merge({
-        location: action.payload,
-      });
-    default:
-      return state;
-  }
-}
-
 /**
  * Creates the main reducer with the dynamically injected ones
  */
 export default function createReducer(injectedReducers) {
-  return combineReducers({
-    route: routeReducer,
-    global: globalReducer,
-    language: languageProviderReducer,
-    ...injectedReducers,
-  });
+  return connectRouter(history)(
```

Comment:
one minor nit: assigning some variables here may help readability... when first reviewing it took me a few seconds to find the `connectRouter` integration

```js
export default function createReducer(injectedReducers) {
  const mergeWithRouterState = connectRouter(history);
  const rootState = combineReducers({
    global: globalReducer,
    language: languageProviderReducer,
    ...injectedReducers,
  });
  return mergeWithRouterState(rootState);
}
```

unrelated, but it might make sense to assign a default parameter for `injectedReducers`...

```js
export default function createReducer(injectedReducers = {}) {
  //...
}
```

or add guard, eg.. 

```js
import isObject from 'lodash/isObject';
import isFunction from 'lodash/isFunction';
import invariant from 'invariant';

export default function createReducer(injectedReducers) {
  invariant(
    isObject(injectedReducers) &&
    Object.keys(injectedReducers) &&
    Object.values(injectedReducers).every(isFunction),
    '(app/reducers) createReducer: `injectedReducers` must be an object of reducer functions'
  )
  // ...
}
```

Reply:
Absolutely to both comments! Thanks!

I went for the default param as combineReducers already has an error message if it doesn't receive an object of reducer functions + using our own guard meant providing an empty object at initial store creation, which felt like a contrived complexity.